dags/roger/core.py

```diff
@@ -782,216 +807,58 @@ def write_schema (self, schema, schema_type: SchemaType):
-        log.info(f"processing {data_set_name} took {total_file_processing_time}")
-        return current_metric
-
     def merge (self):
-        """ Merge nodes. Would be good to have something less computationally intensive. """
+        """ This version uses the disk merging from the kg_utils module """
         data_set_version = self.config.get('kgx', {}).get('dataset_version')
         metrics = {}
         start = time.time()
         json_format_files = Util.kgx_objects("json")
-        jsonl_format_files = set([
-            x.replace(f'nodes_{data_set_version}.jsonl', '').replace(f'edges_{data_set_version}.jsonl', '') for x in Util.kgx_objects("jsonl")
-        ])
-
-        log.info("Deleting any redis merge keys from previous run....")
-        self.delete_all_keys()
-
-        for file in json_format_files:
-            current_metric = {}
-            total_time = read_time = time.time()
-            current_kgx_data = Util.read_object(file)
-            read_time = time.time() - read_time
-            current_metric['read_kgx_file_time'] = read_time
-            current_metric['total_processing_time'] = total_time
-            self.merge_node_and_edges(nodes=current_kgx_data['nodes'],
-                                      edges=current_kgx_data['edges'],
-                                      current_metric=current_metric,
-                                      data_set_name=file)
-
-        for file in jsonl_format_files:
-            current_metric = {}
-            total_time = read_time = time.time()
-            edges = Util.json_line_iter(Util.kgx_path(file + f'edges_{data_set_version}.jsonl'))
-            nodes = Util.json_line_iter(Util.kgx_path(file + f'nodes_{data_set_version}.jsonl'))
-            read_time = time.time() - read_time
-            current_metric['read_kgx_file_time'] = read_time
-            current_metric['total_processing_time'] = total_time
-            self.merge_node_and_edges(nodes=nodes,
-                                      edges=edges,
-                                      current_metric=current_metric,
-                                      data_set_name=file)
-
-        log.info(f"total time for dumping to redis : {time.time() - start}")
-
-        # now we have all nodes and edges merged in redis we scan the whole redis back to disk
+        jsonl_format_files = Util.kgx_objects("jsonl")
+
+        # Create lists of the nodes and edges files in both json and jsonl formats
+        json_node_files = {file for file in json_format_files if "node" in file}
+        jsonl_node_files = {file for file in jsonl_format_files if "node" in file}
+        json_edge_files = {file for file in json_format_files if "edge" in file}
+        jsonl_edge_files = {file for file in jsonl_format_files if "edge" in file}
+
+        # Create all the needed iterators and sets thereof
+        json_node_iterators = [jsonl_iter(file_name, 'nodes') for file_name in json_node_files] 
```

Comment:
i think this would need json_iter instead of jsonl_iter for iterating over json files

Reply:
Cut and paste! Fixed it

Reply:
Still needs more testing though...